src/portable/raspberrypi/rp2040/rp2040_usb.h

```diff
@@ -64,7 +76,11 @@ typedef struct hw_endpoint
 
     // Interrupt, bulk, etc
     uint8_t transfer_type;
-    
+
```

Comment:
since your here, how about moving `active` from above with suitable comment into this group of 1 byte fields... we have 32 of these structures and we were wasting 4 bytes per.

Reply:
```
(gdb) ptype /o struct hw_endpoint
/* offset    |  size */  type = struct hw_endpoint {
/*    0      |     1 */    _Bool configured;
/*    1      |     1 */    _Bool rx;
/*    2      |     1 */    uint8_t ep_addr;
/*    3      |     1 */    uint8_t next_pid;
/*    4      |     4 */    io_rw_32 *endpoint_control;
/*    8      |     4 */    io_rw_32 *buffer_control;
/*   12      |     4 */    uint8_t *hw_data_buf;
/*   16      |     1 */    _Bool active;
/* XXX  1-byte hole  */
/*   18      |     2 */    uint16_t remaining_len;
/*   20      |     2 */    uint16_t xferred_len;
/* XXX  2-byte hole  */
/*   24      |     4 */    uint8_t *user_buf;
/*   28      |     2 */    uint16_t wMaxPacketSize;
/*   30      |     1 */    uint8_t transfer_type;
/*   31      |     1 */    uint8_t pending;

                           /* total size (bytes):   32 */
                         }
```
three according to gdb, but yes holes are bad

Reply:
ah, ok, i thought it would make it smaller, but i was thinking of the 2 uint16_t s as being 4 byte aligned for some reason :-(

Reply:
In host mode the two extra fields end up growing the structure to 36B, so it is worth rearranging so that pad bytes at the end of the structure get used instead.